### ISSUE TITLE
Float at the start of a `position: relative` ancestor protrudes above the page top after a page break

### ISSUE THREAD

**Issue by nagomu** [vivliostyle/vivliostyle.js#1885]
When a float is the first in-flow child of a `position: relative` ancestor, and a page break occurs near that float, the float is rendered **above the top of the page content area** on the next page.

Removing `position: relative` from the ancestor makes the issue disappear.

## Minimal reproduction

Repository: https://github.com/nagomu/vivliostyle-sample20260417

```shell
git clone https://github.com/nagomu/vivliostyle-sample20260417
cd vivliostyle-sample20260417
yarn install
yarn build
# open sample/_page.pdf
```

## Environment

- `@vivliostyle/cli` 10.5.0
- `@vivliostyle/core` 2.41.0

## Expected

The float is rendered inside its container on the next page, below the top of the page content area.

## Actual

The float's top edge is positioned above the top of the page content area (outside the page margin box).

![screenshot](https://raw.githubusercontent.com/nagomu/vivliostyle-sample20260417/main/docs/screen.png)

## Workaround

Removing `position: relative` from the ancestor avoids the bug.

**Comment by nagomu** [comment]
Thank you for all the amazing work on 2.41.0 — the layout improvements are genuinely exciting to see 🥇 
Vivliostyle keeps getting better, and I really appreciate the effort that goes into each release. Hope this report is useful!

**Comment by MurakamiShinyu** [comment]
@nagomu Thanks for reporting the issue!

Fixed:
- PR #1886 

Test: https://vivliostyle.vercel.app/#src=https://raw.githubusercontent.com/nagomu/vivliostyle-sample20260417/refs/heads/main/sample/page.html

**Comment by nagomu** [comment]
Thanks 🙏
